libs/core/src/main/java/org/elasticsearch/core/RefCounted.java

```diff
@@ -29,6 +29,10 @@
  */
 public interface RefCounted {
 
+    default int refCnt() {
```

Comment:
Is this really necessary? Can we just make `ByteBufRefCounted` default visible with this method for the test instead of adding something that isn't going to work for the majority of implementations?

Reply:
Not necessary, I think it's odd to have RefCounted without refCnt method. But it's outside of this change. Made it local to Netty4Utils.

Reply:
I guess I kind of agree. It's a common method for this type of thing, which is why if we were to add it to the interface I would prefer we actually make it work for everything. Instead of an expected method being there, but not actually working in a lot of cases.